Lib/test/test_itertools.py

```diff
@@ -1976,6 +1976,14 @@ def gen2(x):
         self.assertRaises(AssertionError, list, cycle(gen1()))
         self.assertEqual(hist, [0,1])
 
+    def test_deep_recursion(self):
```

Comment:
Now this is not a recursion. The recursion from user side is `chain.from_iterable(chain.from_iterable(...))`. Maybe use other name for the test?